extensions/spotify-player/src/components/NowPlayingDetail.tsx

```diff
@@ -0,0 +1,49 @@
+import { Icon, List } from "@raycast/api";
+import { useEffect, useState } from "react";
+import { SpotifyPlayingState, SpotifyState, TrackInfo } from "../spotify/types";
+import { msToHMS } from "../utils";
+
+export default function NowPlayingDetail({
+  trackInfo,
+  playerState,
+}: {
+  trackInfo: TrackInfo;
+  playerState: SpotifyState;
+}): JSX.Element {
+  const [position, setPosition] = useState(playerState.position * 1000);
+  const trackDuration = trackInfo.duration;
+
+  useEffect(() => {
+    setPosition(playerState.position * 1000);
+  }, [playerState.position]);
+
+  useEffect(() => {
+    const id = setTimeout(() => {
+      if (playerState.state === SpotifyPlayingState.Playing && position < trackDuration) {
+        setPosition(position + 1000);
+      }
+    }, 1000);
+
+    return () => {
+      clearTimeout(id);
+    };
+  }, [position]);
+
+  return (
+    <List.Item.Detail
+      markdown={`<img src="${trackInfo.artwork_url}" alt="${trackInfo.album}]" width="160" height="160" />`}
```

Comment:
This seems to be the source of the issue around images sometimes not being loaded correctly but there isnt really much to this so I assume it must be the Raycast proxy rate limiting the user?